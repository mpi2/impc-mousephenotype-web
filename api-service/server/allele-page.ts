import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { AlleleSummary } from "@/models";
const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchAlleleSummary(
  mgiGeneAccessionId: string,
  alleleName: string,
): Promise<AlleleSummary> {
  const endpointURL = `http://impc-allele-service.${KUBERNETES_NAMESPACE}:8080/v1/alleles?mgiGeneAccessionId=${mgiGeneAccessionId}&alleleName=${alleleName}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<AlleleSummary>(
        `/api/v1/alleles/${mgiGeneAccessionId}/${alleleName}`,
      )
    : fetchURL<AlleleSummary>(endpointURL));
}
