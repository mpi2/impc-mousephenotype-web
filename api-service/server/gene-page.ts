import { GeneOrder, GenePhenotypeHits, GeneSummary } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { processGeneOrderResponse } from "@/hooks/gene-order.query";
import { processGenePhenotypeHitsResponse } from "@/hooks/significant-phenotypes.query";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchGeneSummary(
  mgiGeneAccessionId: string,
): Promise<GeneSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/gene_summary?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<GeneSummary>(
        `/api/v1/genes/${mgiGeneAccessionId}/summary`,
      )
    : fetchURL<GeneSummary>(endpointURL));
}

export async function fetchGenePhenotypeHits(
  mgiGeneAccessionId: string,
): Promise<Array<GenePhenotypeHits>> {
  const endpointURL = `http://impc-phenotype-hits-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotypehits?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GenePhenotypeHits>>(
        `/api/v1/genes/${mgiGeneAccessionId}/phenotype-hits`,
      )
    : fetchURL<Array<GenePhenotypeHits>>(endpointURL));
  return processGenePhenotypeHitsResponse(data);
}

export async function fetchGeneOrderData(
  mgiGeneAccessionId: string,
): Promise<Array<GeneOrder>> {
  const endpointURL = `http://impc-order-service.${KUBERNETES_NAMESPACE}:8080/v1/orders?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GeneOrder>>(
        `/api/v1/genes/${mgiGeneAccessionId}/order`,
      )
    : fetchURL<Array<GeneOrder>>(endpointURL));
  return processGeneOrderResponse(data);
}
