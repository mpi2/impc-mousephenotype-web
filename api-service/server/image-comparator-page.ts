import { GeneImageCollection } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchMutantImages(
  mgiGeneAccessionId: string,
  parameterStableId: string,
): Promise<Array<GeneImageCollection>> {
  const endpointURL = `http://impc-imaging-service.${KUBERNETES_NAMESPACE}:8080/v1/images/findByMgiAndStableId?mgiGeneAccessionId=${mgiGeneAccessionId}&parameterStableId=${parameterStableId}`;
  try {
    return await (NODE_ENV === "development"
      ? fetchAPIFromServer<Array<GeneImageCollection>>(
          `/api/v1/images/find_by_mgi_and_stable_id?mgiGeneAccessionId=${mgiGeneAccessionId}&parameterStableId=${parameterStableId}`,
        )
      : fetchURL<Array<GeneImageCollection>>(endpointURL));
  } catch (error) {
    return [];
  }
}

export async function fetchControlImages(
  parameterStableId: string,
): Promise<Array<GeneImageCollection>> {
  const endpointURL = `http://impc-imaging-service.${KUBERNETES_NAMESPACE}:8080/v1/images/findByStableIdAndSampleId?biologicalSampleGroup=control&parameterStableId=${parameterStableId}`;
  try {
    return await (NODE_ENV === "development"
      ? fetchAPIFromServer<Array<GeneImageCollection>>(
          `/api/v1/images/find_by_stable_id_and_sample_id?biologicalSampleGroup=control&parameterStableId=${parameterStableId}`,
        )
      : fetchURL<Array<GeneImageCollection>>(endpointURL));
  } catch (error) {
    return [];
  }
}
