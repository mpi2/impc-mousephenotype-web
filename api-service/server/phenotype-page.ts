import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { PhenotypeSummary } from "@/models/phenotype";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchPhenotypeSummary(
  phenotypeId: string,
): Promise<PhenotypeSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotype_summary?phenotypeId=${phenotypeId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<PhenotypeSummary>(
        `/api/v1/phenotypes/${phenotypeId}/summary`,
      )
    : fetchURL<PhenotypeSummary>(endpointURL));
}
