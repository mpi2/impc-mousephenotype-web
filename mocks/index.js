export async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen();
  } else {
    const { worker } = await import("./browser");
    return worker.start({
      serviceWorker: {
        url: "/data/mockServiceWorker.js",
        options: {
          scope: "/",
        }
      }
    });
  }
}

