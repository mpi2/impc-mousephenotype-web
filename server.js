const { createServer } = require('http');
const { parse } = require('url');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const hostname = 'localhost';
const port = 3000;
const app = next({ dev, hostname, port });
const handle = app.getRequestHandler();

app.prepare()
.then(() => {
    createServer(async (req, res) => {
        try {
            const parsedUrl = parse(req.url, true);
            const { pathname, query} = parsedUrl;
            const host = req.headers.host;
            console.log({ host, pathname });
            res.setHeader('X-IMPC-TEST', "true");
            if (host.includes('nginx.mousephenotype-dev.org')) {
                const modifiedPathName = pathname.replace("/data/", "");
                console.log({ modifiedPathName });
                await app.render(req, res, modifiedPathName, query);
                res.setHeader('X-PATH', JSON.stringify({ host, pathname, modifiedPathName }));
            } else {
                await handle(req, res, parsedUrl);
            }

        } catch (err) {
            console.error('Error occurred handling', req.url, err)
            res.statusCode = 500
            res.end('internal server error')
        }
    })
        .once('error', (err) => {
            console.error(err);
            process.exit(1);
        })
        .listen(port, () => {
            console.log(`Ready on http://${hostname}:${port}`);
        })
});