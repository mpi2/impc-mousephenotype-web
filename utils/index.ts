import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};
