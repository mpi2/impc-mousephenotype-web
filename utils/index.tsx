import React from "react";
import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};

export const formatPValue = (pValue: number) => {
  const pValueArray = Number.parseFloat(String(pValue))
    .toExponential(2)
    .split("e");
  return (
    <>
      {pValueArray[0]}x10<sup>{pValueArray[1].replace("+", "")}</sup>
    </>
  );
};

export const formatESCellName = (src: string) => {
  const [name, superscript, end] = src.split(/<|>/);
  return (
    <>
      {name}
      <sup>{superscript}</sup>
      {end}
    </>
  );
};

const fetchCache = {};
export { fetchCache };

export const toSentenceCase = (camelCase) => {
  if (camelCase) {
    const result = camelCase.replace(/([A-Z])/g, " $1");
    return result[0].toUpperCase() + result.substring(1).toLowerCase();
  }
  return "";
};
