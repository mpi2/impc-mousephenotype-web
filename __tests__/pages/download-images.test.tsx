import { screen } from '@testing-library/react';
import DownloadImagesPage from "@/pages/genes/[pid]/download-images/[parameterStableId]";
import { API_URL, renderWithClient } from "../utils";
import mockRouter from "next-router-mock";
import { waitFor } from "@testing-library/react";
import { server } from "../../mocks/server";
import { rest } from "msw";
import dbn1MutantImages from '../../mocks/data/tests/dbn1-mutant-dl-images.json';
import dbn1ControlImages from '../../mocks/data/tests/dbn1-control-dl-images.json';

jest.mock('next/router', () => jest.requireActual('next-router-mock'));

describe('Download images page', () => {
  it('renders correctly', async () => {
    server.use(
      rest.get(`${API_URL}/api/v1/images/find_by_mgi_and_stable_id?mgiGeneAccessionId=MGI:1931838&parameterStableId=JAX_ERG_028_001`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.json(dbn1MutantImages));
      })
    );
    server.use(
      rest.get(`${API_URL}/api/v1/images/find_by_stable_id_and_sample_id?biologicalSampleGroup=control&parameterStableId=JAX_ERG_028_001`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.json(dbn1ControlImages));
      })
    );
    await mockRouter.push('data/genes/MGI:1931838/download-images/JAX_ERG_028_001?pid=MGI:1931838&parameterStableId=JAX_ERG_028_001');
    const { container } = renderWithClient(
      <DownloadImagesPage />
    );
    await waitFor(() => screen.getByRole('heading', { level: 1 }));
    expect(container).toMatchSnapshot();
  });
});