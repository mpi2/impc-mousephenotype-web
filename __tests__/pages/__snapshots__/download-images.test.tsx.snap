// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download images page renders correctly 1`] = `
<div>
  <div
    class="banner"
  >
    <div
      class="pb-4 pt-5 container container"
    >
      <div
        class="col-12 col-md-8 ps-4 pe-4"
      >
        <div
          class="tabs"
        >
          <a
            class="tab__active"
            href="/search"
          >
            Genes
          </a>
          <a
            class="tab"
            href="/search?type=phenotype"
          >
            Phenotypes
          </a>
          <a
            class="tab"
            href="https://dev.mousephenotype.org/?s="
          >
            Help, news, blog
          </a>
        </div>
        <div
          class="inputCont"
        >
          <input
            class="input"
            placeholder="Search for a gene..."
            title="main search box"
            type="text"
            value=""
          />
          <button
            aria-describedby="svg-inline--fa-title-search-icon"
            class="searchBtn"
          >
            <svg
              aria-labelledby="svg-inline--fa-title-search-icon"
              class="svg-inline--fa fa-magnifying-glass "
              data-icon="magnifying-glass"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-search-icon"
              >
                Search button
              </title>
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page container"
  >
    <div
      class="card"
    >
      <a
        class="grey mb-3 small"
        href="/genes/MGI:1931838#images"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-left-long "
          data-icon="arrow-left-long"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 288 480 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-370.7 0 73.4-73.4c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-128 128z"
            fill="currentColor"
          />
        </svg>
          BACK TO GENE
      </a>
      <h1
        class="mb-4 mt-2"
        style="display: flex; gap: 1rem;"
      >
        <strong>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="width: 50px;"
            >
              ‌
            </span>
          </span>
        </strong>
         
        / 
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="width: 50px;"
          >
            ‌
          </span>
        </span>
      </h1>
      <h2>
        Mutant Files
      </h2>
      <h2
        class="mt-2"
      >
        Control Files
      </h2>
    </div>
  </div>
</div>
`;
